### PR TITLE
Added code from udacity to unit tests

### PR DESCRIPTION
**Pull request by iamtrask**

### CODE REVIEW

File: test/test_udacity.py [hunk 1]
```diff
@@ -0,0 +1,836 @@
...
+    y
+
+    x = th.tensor([0.1, 0.2, 0.3])
+
+    x = x.fix_prec().share(bob, alice, secure_worker)
+
+    y = x + x
+
+    y.get().float_prec()
+
```
Comment (robert-wagner) [inline test/test_udacity.py]: Delete these blank lines

File: test/test_udacity.py [hunk 2]
```diff
@@ -0,0 +1,836 @@
...
+    x
+
+    bob._objects
+
+    alice._objects
+
+    assert True
+
+
+def test_section_3_securing_fl(hook):
```
Comment (robert-wagner) [inline test/test_udacity.py]: Should you have assertions in this code

File: test/test_udacity.py [hunk 3]
```diff
@@ -0,0 +1,836 @@
...
+
+    x.move(bob)
+
+    x
+
+    bob._objects
+
+    alice._objects
+
+    assert True
```
Comment (robert-wagner) [inline test/test_udacity.py]: Never assert true. It's a pointless line

File: test/test_udacity.py [hunk 4]
```diff
@@ -0,0 +1,836 @@
...
+
+    alice._objects
+
+    x.move(bob)
+
+    x
+
+    bob._objects
+
+    alice._objects
```
Comment (robert-wagner) [inline test/test_udacity.py]: should there be assertions about the contents of these objects

Comment (iamtrask) [reply test/test_udacity.py]: no - i'm just copy pasting the whole notebook - i'd rather not make edits

File: test/test_udacity.py (hunk 3 above)
Comment (iamtrask) [reply test/test_udacity.py]: k

File: test/test_udacity.py (hunk 2 above)
Comment (iamtrask) [reply test/test_udacity.py]: no - it's just checking that the code runs

File: test/test_udacity.py (hunk 1 above)
Comment (iamtrask) [reply test/test_udacity.py]: they're separate cells in the notebook - at some point i'm going to have to come back and find bits of code to change,. the lines help somee

**Comment by mari-linhares** [comment]
Why are polynomial tensor files commented?

**Comment by iamtrask** [comment]
> Why are polynomial tensor files commented?

The test is failing in very unexpected ways (some kind of segfault which I couldn't track down).

Also, the rest of the codebase isn't yet using it, and it likely needs significant refactoring before the reset of the codebase can use it. In particular, it's not yet actually a tensor type, it's a class with static functions. We can revisit it after the course is released.

**Pull request merged**